CRM/MembershipExtras/Job/OfflineAutoRenewal/MultipleInstallmentPlan.php

```diff
@@ -163,8 +183,7 @@ private function calculateNewPeriodStartDate() {
       return $latestDate->format('Y-m-d');
     }
 
-    $installmentReceiveDateCalculator = new InstallmentReceiveDateCalculator($this->currentRecurringContribution);
-    return $installmentReceiveDateCalculator->calculate($this->currentRecurringContribution['installments'] + 1);
+    return $this->calculateNewPeriodStartDate();
```

Comment:
I think  since u already calculated this before and assigned it to this->paymentPlanStartDate, then here you can just do :

```
return this->paymentPlanStartDate;
```

Reply:
I did something similar. Didn't think it was a good idea to couple those two methods, so instead I'm returning NULL on default, and handled it by checking if it returned NULL assigning $this->paymentPlanStartDate to memberships start date.